core/tauri-build/src/acl.rs

```diff
@@ -28,43 +29,118 @@ const CAPABILITIES_SCHEMA_FILE_NAME: &str = "schema.json";
 /// Path of the folder where schemas are saved.
 const CAPABILITIES_SCHEMA_FOLDER_PATH: &str = "gen/schemas";
 const CAPABILITIES_FILE_NAME: &str = "capabilities.json";
-const PLUGIN_MANIFESTS_FILE_NAME: &str = "plugin-manifests.json";
+const ACL_MANIFESTS_FILE_NAME: &str = "acl-manifests.json";
+
+/// Definition of a plugin that is part of the Tauri application instead of having its own crate.
+///
+/// By default it generates a plugin manifest that parses permissions from the `permissions/$plugin-name` directory.
+/// To change the glob pattern that is used to find permissions, use [`Self::permissions_path_pattern`].
+///
+/// To autogenerate permissions for each of the plugin commands, see [`Self::commands`].
+#[derive(Debug, Default)]
+pub struct InlinedPlugin {
+  commands: &'static [&'static str],
+  permissions_path_pattern: Option<&'static str>,
+}
+
+impl InlinedPlugin {
+  pub fn new() -> Self {
+    Self::default()
+  }
+
+  /// Define a list of commands that gets permissions autogenerated in the format of `allow-$command` and `deny-$command`
+  /// where $command is the command name in snake_case.
+  pub fn commands(mut self, commands: &'static [&'static str]) -> Self {
+    self.commands = commands;
+    self
+  }
+
+  /// Sets a glob pattern that is used to find the permissions of this inlined plugin.
+  ///
+  /// **Note:** You must emit [rerun-if-changed] instructions for the plugin permissions directory.
+  ///
+  /// By default it is `./permissions/$plugin-name/**/*`
+  pub fn permissions_path_pattern(mut self, pattern: &'static str) -> Self {
+    self.permissions_path_pattern.replace(pattern);
+    self
+  }
+}
+
+/// Tauri application permission manifest.
+///
+/// By default it generates a manifest that parses permissions from the `permissions` directory.
+/// To change the glob pattern that is used to find permissions, use [`Self::permissions_path_pattern`].
+///
+/// To autogenerate permissions for each of the app commands, see [`Self::commands`].
+#[derive(Debug, Default)]
+pub struct AppManifest {
+  commands: &'static [&'static str],
+  permissions_path_pattern: Option<&'static str>,
+}
+
+impl AppManifest {
+  pub fn new() -> Self {
+    Self::default()
+  }
+
+  /// Define a list of commands that gets permissions autogenerated in the format of `allow-$command` and `deny-$command`
+  /// where $command is the command name in snake_case.
+  pub fn commands(mut self, commands: &'static [&'static str]) -> Self {
+    self.commands = commands;
+    self
+  }
+
+  /// Sets a glob pattern that is used to find the permissions of the app.
+  ///
+  /// **Note:** You must emit [rerun-if-changed] instructions for the permissions directory.
```

Comment:
can't we do that internally?

Reply:
if you set a custom glob pattern it's hard to determine which path should be used to rerun-if-changed

Reply:
makes sense